swimster-ui/src/hooks/useReservationCard.js

```diff
@@ -1,127 +1,89 @@
-                const numbersBetween = inclusiveInterval(formatted_start_time,  formatted_end_time)
-                reservedTimesObj[formatted_reservation_date].push(...numbersBetween)
-            }
-        })
-        return reservedTimesObj
-    }
+    const isAuthenticated = isUserAuthenticated(user, initialized)
 
-    // sets the registered reservation fields' values for form submission
-    const handleReservations = () => {
-        setValue("reservationDate", toLocalTime(dateSelected))
-        setValue("reservationStartTime", format(startTime, "HH:mm:ss"))
-        setValue("reservationEndTime", format(endTime, "HH:mm:ss"))
-    }
+    const { reservedTimes, setReservedTimes, checkSingleTimes, checkReservedTimes, toLocalTime } = useReserve({ reservation })
 
     // updates the time fields with the corresponding reserved times when user selects a date
     useEffect(() => {
-        const result = checkReservedTimes(reservations)
+        const result = checkReservedTimes(fetchedReservations)
         const reservedTimesWithNoGaps = checkSingleTimes(result)
         setReservedTimes(reservedTimesWithNoGaps)
-    }, [dateSelected, reservations])
+    }, [dateSelected, fetchedReservations])
+
+    // updates date and times for every selection change user makes
+    useEffect(() => {
+        setValue("reservationDate", toLocalTime(dateSelected))
+        if (startTime !== "") setValue("startTime", format(startTime, "HH:mm:ss"))
+        if (endTime !== "") setValue("endTime", format(endTime, "HH:mm:ss"))
+    }, [dateSelected, startTime, endTime])
 
+    // fetches reservations from listing on first user click
     useEffect(() => {
         const fetchReservationsFromListing = async () => {
             const { data, error } = await apiClient.fetchReservationsFromListing(listingId)
             if (error) setError(error)
-            if (data?.reservationsForListing) setReservations([...data.reservationsForListing])
+            if (data?.reservationsForListing) setFetchedReservations([...data.reservationsForListing])
         }
 
         if (isAuthenticated) fetchReservationsFromListing()
     }, [listingId, isAuthenticated])
 
+    // handler for making a reservation for the current listing
+    const handleReservation = async formData => {
+        setIsSubmitProcessing(true)
+
+        const formattedFormData = {
+            reservationDate: formData.reservationDate,
+            startTime: formData.startTime,
+            endTime: formData.endTime,
+            guests: parseInt(formData.reservationGuests),
+        }
+
+        const { data, error } = await apiClient.createReservation(JSON.stringify(formattedFormData), listingId)
+        if (error) setError(error)
+        if (data?.reservation) {
+            setReservation(data.reservation)
+            navigate('confirm')
+        }
+
+        setIsSubmitProcessing(false)
```

Comment:
it seems that you arent using IsSubmitProcessing anymore? is there a need for it now?

Reply:
I will use loading to navigate to the confirmation page. There will be a PR for loading very soon.